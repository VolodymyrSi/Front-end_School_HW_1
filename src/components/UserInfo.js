import PropTypes from "prop-types";

const UserInfo = ({ user, stats }) => {
  return (
    <div className="bg-dark text-light d-flex align-items-center justify-content-center p-3">
      {user && (
        <div>
          <h2 className="text-center">{user.nickname}</h2>
          <img alt={user.nickname} className="w-200" src={user.avatarLarger} />
        </div>
      )}
      {stats && (
        <div className="ml-3">
          <p>{`‚¨ÖÔ∏è Followers: ${stats.followerCount.toLocaleString()}`}</p>
          <p>{`‚û°Ô∏è Following: ${stats.followingCount.toLocaleString()}`}</p>
          <p>{`‚ù§Ô∏è Total likes: ${stats.heartCount.toLocaleString()}`}</p>
          <p>{`üé• Total videos: ${stats.videoCount.toLocaleString()}`}</p>
        </div>
      )}
    </div>
  );
};

UserInfo.propTypes = {
  stats: PropTypes.arrayOf(
    PropTypes.shape({
      followerCount: PropTypes.string.isRequired,
      followingCount: PropTypes.string.isRequired,
      heartCount: PropTypes.string.isRequired,
      videoCount: PropTypes.string.isRequired,
    })
  ),
  user: PropTypes.arrayOf(
    PropTypes.shape({
      avatarLarger: PropTypes.string.isRequired,
      nickname: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default UserInfo;
