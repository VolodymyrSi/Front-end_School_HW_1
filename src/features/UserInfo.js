import PropTypes from "prop-types";

const UserInfo = ({ user, stats }) => {
  return (
    <div className="bg-dark text-light d-flex align-items-center justify-content-center p-3">
      {user && (
        <div>
          <h2 className="text-center">{user.nickname}</h2>
          <img alt={user.nickname} className="w-200" src={user.avatarLarger} />
        </div>
      )}
      {stats && (
        <div className="ml-3">
          <p>{`‚¨ÖÔ∏è Followers: ${stats.followerCount.toLocaleString()}`}</p>
          <p>{`‚û°Ô∏è Following: ${stats.followingCount.toLocaleString()}`}</p>
          <p>{`‚ù§Ô∏è Total likes: ${stats.heartCount.toLocaleString()}`}</p>
          <p>{`üé• Total videos: ${stats.videoCount.toLocaleString()}`}</p>
        </div>
      )}
    </div>
  );
};

UserInfo.propTypes = {
  stats: PropTypes.shape({
    followerCount: PropTypes.string.isRequired,
    followingCount: PropTypes.string.isRequired,
    heartCount: PropTypes.string.isRequired,
    videoCount: PropTypes.string.isRequired,
  }),
  user: PropTypes.shape({
    avatarLarger: PropTypes.string.isRequired,
    nickname: PropTypes.string.isRequired,
  }),
};

UserInfo.defaultProps = {
  stats: {
    followerCount: "no data",
    followingCount: "no data",
    heartCount: "no data",
    videoCount: "no data",
  },
  user: {
    nickname: "John Doe",
    avatarLarger:
      "https://pbs.twimg.com/profile_images/1326707048478892033/Ln0v50LP_400x400.jpg",
  },
};

export default UserInfo;
