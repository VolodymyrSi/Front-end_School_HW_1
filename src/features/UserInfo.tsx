interface UserTypes {
  nickname: string;
  avatarLarger: string;
}

interface StatsTypes {
  followerCount: string;
  followingCount: string;
  heartCount: string;
  videoCount: string;
}

interface UserInfoProperties {
  user?: UserTypes;
  stats?: StatsTypes;
}

const UserInfo = ({ user, stats }: UserInfoProperties) => {
  return (
    <div className="bg-dark text-light d-flex align-items-center justify-content-center p-3 flex-column flex-md-row">
      {user && (
        <div className="text-center">
          <h2 className="text-center">{user.nickname}</h2>
          <img
            alt={user.nickname}
            className="w-50 mb-2"
            src={user.avatarLarger}
          />
        </div>
      )}
      {stats && (
        <div className="ml-3">
          <p>{`‚¨ÖÔ∏è Followers: ${stats.followerCount.toLocaleString()}`}</p>
          <p>{`‚û°Ô∏è Following: ${stats.followingCount.toLocaleString()}`}</p>
          <p>{`‚ù§Ô∏è Total likes: ${stats.heartCount.toLocaleString()}`}</p>
          <p>{`üé• Total videos: ${stats.videoCount.toLocaleString()}`}</p>
        </div>
      )}
    </div>
  );
};

UserInfo.defaultProps = {
  stats: {
    followerCount: "I wish I could access a server",
    followingCount: "I remember I could do things like that",
    heartCount: "I do not feel connected...",
    videoCount: "This must be some sort of punishment. Am I in hell?",
  },
  user: {
    nickname: "John Doe",
    avatarLarger:
      "https://pbs.twimg.com/profile_images/1326707048478892033/Ln0v50LP_400x400.jpg",
  },
};

export default UserInfo;
