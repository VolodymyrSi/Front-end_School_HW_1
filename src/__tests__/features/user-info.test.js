import Enzyme, { shallow } from "enzyme";
import UserInfo from "../../features/UserInfo";
import Adapter from "enzyme-adapter-react-16";
import "jest-enzyme";

Enzyme.configure({ adapter: new Adapter() });

const user = {
  nickname: "Alice Woo",
  avatarLarger:
    "https://i.pinimg.com/474x/56/8f/a2/568fa2f17d494f297e609eb5e9404c8d.jpg",
};

const stats = {
  followerCount: "1",
  followingCount: "2",
  heartCount: "3",
  videoCount: "4",
};

describe("UserInfo tests", () => {
  it("should contain .d-flex wrapper", function () {
    const component = shallow(<UserInfo user={user} stats={stats} />);
    const wrapper = component.find(".d-flex");
    expect(wrapper.length).toBe(1);
    console.log(component.debug());
  });
  it("should contain an image", function () {
    const component = shallow(<UserInfo user={user} stats={stats} />);
    const wrapper = component.find("img");
    expect(wrapper.length).toBe(1);
    console.log(component.debug());
  });
  it("should render userInfo with props", function () {
    const component = shallow(<UserInfo user={user} stats={stats}/>);
    expect(component.debug()).toMatchSnapshot()
  });
  it("should render userInfo without props", function () {
    const component = shallow(<UserInfo/>);
    expect(component.debug()).toMatchSnapshot()
  });
});
