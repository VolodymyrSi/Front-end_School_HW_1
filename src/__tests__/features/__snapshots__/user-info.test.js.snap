// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo tests should render userInfo with props 1`] = `
"<div className=\\"bg-dark text-light d-flex align-items-center justify-content-center p-3\\">
  <div>
    <h2 className=\\"text-center\\">
      Alice Woo
    </h2>
    <img alt=\\"Alice Woo\\" className=\\"w-200\\" src=\\"https://i.pinimg.com/474x/56/8f/a2/568fa2f17d494f297e609eb5e9404c8d.jpg\\" />
  </div>
  <div className=\\"ml-3\\">
    <p>
      ‚¨ÖÔ∏è Followers: 1
    </p>
    <p>
      ‚û°Ô∏è Following: 2
    </p>
    <p>
      ‚ù§Ô∏è Total likes: 3
    </p>
    <p>
      üé• Total videos: 4
    </p>
  </div>
</div>"
`;

exports[`UserInfo tests should render userInfo without props 1`] = `
"<div className=\\"bg-dark text-light d-flex align-items-center justify-content-center p-3\\">
  <div>
    <h2 className=\\"text-center\\">
      John Doe
    </h2>
    <img alt=\\"John Doe\\" className=\\"w-200\\" src=\\"https://pbs.twimg.com/profile_images/1326707048478892033/Ln0v50LP_400x400.jpg\\" />
  </div>
  <div className=\\"ml-3\\">
    <p>
      ‚¨ÖÔ∏è Followers: no data
    </p>
    <p>
      ‚û°Ô∏è Following: no data
    </p>
    <p>
      ‚ù§Ô∏è Total likes: no data
    </p>
    <p>
      üé• Total videos: no data
    </p>
  </div>
</div>"
`;
